xcode/Subconscious/Shared/Components/AddressBook/AddressBookModel.swift

```diff
@@ -209,17 +230,40 @@ struct AddressBookModel: ModelProtocol {
             model.isFollowUserFormPresented = false
             model.follows.append(entry)
             
-            return Update(state: model)
+            return update(
+                state: model,
+                action: .refreshEntries(forceRefetchFromNoosphere: true),
+                environment: environment
+            )
             
         case .failFollow(error: let error):
             var model = state
             model.failFollowErrorMessage = error
             return Update(state: model)
             
+        case .dismissFailFollowError:
+            var model = state
+            model.failFollowErrorMessage = nil
+            return Update(state: model)
+            
         case .requestUnfollow(petname: let petname):
+            var model = state
+            model.unfollowCandidate = petname
+            return Update(state: model)
+            
+        case .cancelUnfollow:
+            var model = state
+            model.unfollowCandidate = nil
+            return Update(state: model)
+            
+        case .confirmUnfollow:
+            guard let petname = state.unfollowCandidate else {
+                return Update(state: state)
+            }
+            
             let fx: Fx<AddressBookAction> =
                 environment.data.addressBook
-                .unsetPetnameAsync(petname: petname)
+                .unfollowUserAsync(petname: petname)
```

Comment:
Higher-level concept: following is user-level and petname is unset is an implementation detail